processing-service-core/tenant-config-options.md

```diff
@@ -0,0 +1,200 @@
+
+### Bulk retrieval of tenant configuration 
+All of the tenant specific configurations that have been set for a specific tenant call be return in on call to the api. 
+This bulk request returns an array of tenant configs that are assigned to the tenantId provided.
+
+A sample bulk get request is as follows:
+
+        http://<host>:<port>/data-processing-service/v1/tenants/{tenantId}/config
+
+A sample response is below:
+
+        {
+            "configurations": [
+            {
+            "key": "CustomTenantConfiguration_1",
+            "value": "customTenantConfigValue"
+            },
+            {
+            "key": "CustomTenantConfiguration_2",
+            "value": "customTenantConfigValue"
+            },
+            {
+            "key": "CustomTenantConfiguration_3",
+            "value": "customTenantConfigValue"
+            }
+            ]
+        }   
+
+When successful this call will return a 200 response with a response body similar to that above. In a failure senario, like when the tenant has no custom configuration, then a 404 response status will be returned along with a message that states no configuration could be found.
+
+## Deleting tenant configuration
+
+After custom configuration has been set for a tenant that configuration can then be deleted, this, like the other operations available from this service can either be done in a per configuration or bulk way.
+
+### Per config deletion
+
+To delete a specific configuration for a specific tenant the service will require the following information:
+
+1. **tenantId** - The id of the tenant that should be used when removing the configuration.
+2. **key** - the key of the configuration that should be removed.
+
+A sample per config delete request is as follows:
+
+        http://<host>:<port>/data-processing-service/v1/tenants/{tenantId}/config/{key}
+
+When this call completes successfully it will return a 204 status code with no content, if no configuration can be found either for that key against that tenant or for that tenant at all, a 404 message is return along with a suitable message.
+
+### Bulk config deletion
+
+A call to delete all configuration registered for a tenant can be made using this api. This will remove any custom configuration that has been set for the tenant.
+
+A sample bulk delete request is as follows:
+
+        http://<host>:<port>/data-processing-service/v1/tenants/{tenantId}/config
+
+When this completes successfully it will return a 204 status code with no content. If no configuration could be found for this tenant then a 404 response status will be return with a suitable error message.
+
+## Get Effective config
+
+A tenant can only set a custom configuration value for a configuration that already has a corresponding global configuration. The global configuration contains a default value for this configuration to be used in cases where it is request for a tenant that has set no custom value.
```

Comment:
'A tenant can only set a custom configuration value', you're using 'custom' in place of 'tenant' configuration here. We should stick on a single name to refer to these things as for consistency.

'where it is request for a tenant', should be 'where it is *requested* for a tenant'.